app/routes/api/v2/OauthRoute.js

```diff
@@ -0,0 +1,24 @@
+import { isEmpty } from 'lodash';
+
+import OauthController from '../../../controllers/api/v2/OauthController';
+import { load as configLoader } from '../../../../config/config';
+
+const oauthConfig = configLoader().oauth || {};
+
+export default function addRoutes(app) {
+  if (!isEmpty(oauthConfig.facebookClientId)) {
+    app.get('/v2/oauth/facebook', OauthController.facebook);
+    app.get('/v2/oauth/facebook/callback', OauthController.facebookCallback);
```

Comment:
Probably it is better to use common patterns like '/v2/oauth/:provider' and '/v2/oauth/:provider/callback' here and check config in controller methods?

Reply:
This will require a refactoring. Working on it.

Reply:
Done.